pkg/operator/ceph/spec/spec.go

```diff
@@ -367,11 +367,13 @@ func generateLifeCycleCmd(dataDirHostPath string) []string {
 func PodLifeCycle(dataDirHostPath string) *v1.Lifecycle {
 	cmd := generateLifeCycleCmd(dataDirHostPath)
 
-	return &v1.Lifecycle{
+	lifeCyleHooks := &v1.Lifecycle{
```

Comment:
Was the lifecycle hook not needed since you added the release after the ceph-osd process exits? Is this change still needed?

Reply:
Yes. `preStop` life cycle hook would be more relevant for fix in #3755. Reverted these changes back to original.